app/registries/foreman/plugin.rb

```diff
@@ -345,8 +345,8 @@ def add_all_permissions_to_default_roles
     def pending_migrations
       return true if Foreman.in_setup_db_rake?
       migration_paths = ActiveRecord::MigrationContext.new(
-        ActiveRecord::Migrator.migrations_paths).migrations
-      pending_migrations = ActiveRecord::Migrator.new(:up, migration_paths).
+        ActiveRecord::Migrator.migrations_paths, ActiveRecord::SchemaMigration).migrations
+      pending_migrations = ActiveRecord::Migrator.new(:up, migration_paths, ActiveRecord::SchemaMigration).
         pending_migrations
```

Comment:
Do we really need all this code? `ActiveRecord::Base.connection.migration_context.needs_migration?` will give the same return value afaict, we would lose the debug logging but i highly doubt it's actually useful at all

Reply:
here you go:
https://github.com/theforeman/foreman/pull/7571